pkg/datapath/loader/base.go

```diff
@@ -346,6 +346,13 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 		args[initArgTunnelMode] = option.TunnelVXLAN
 	}
 
+	if option.Config.Tunnel == option.TunnelDisabled &&
+		option.Config.EnableNodePort &&
+		option.Config.NodePortMode == option.NodePortModeDSR &&
+		option.Config.LoadBalancerDSRDispatch == option.DSRDispatchGeneve {
+		args[initArgTunnelMode] = option.TunnelGeneve
```

Comment:
What if the tunnel mode is set to `vxlan`? Seems like this would stay as "vxlan" and geneve dispatch wouldn't work?

(To be clear, I'd rather catch this way earlier during the pkg/option config parsing, not here)

Reply:
VXLAN with DSR doesn't work. We check it here.
https://github.com/cilium/cilium/pull/23890/files#diff-c1e209552c5426ba2f2e01f043275558eb9f8ea65553210d2f9808de17b59d1eR188-R199